models/unidex/optimism/unidex_optimism_trades.sql

```diff
@@ -0,0 +1,117 @@
+    dexs.block_time,
+    erc20a.symbol                                          AS token_bought_symbol,
+    erc20b.symbol                                          AS token_sold_symbol,
+    CASE
+        WHEN lower(erc20a.symbol) > lower(erc20b.symbol)
+        THEN concat(erc20b.symbol, '-', erc20a.symbol)
+        ELSE concat(erc20a.symbol, '-', erc20b.symbol)
+    END                                                  AS token_pair,
+    dexs.token_bought_amount_raw / power(10, erc20a.decimals) AS token_bought_amount,
+    dexs.token_sold_amount_raw / power(10, erc20b.decimals)   AS token_sold_amount,
+    dexs.token_bought_amount_raw                              AS token_bought_amount_raw,
+    dexs.token_sold_amount_raw                               AS token_sold_amount_raw,
+    COALESCE(
+        dexs.amount_usd,
+        (dexs.token_bought_amount_raw / power(10, erc20a.decimals)) * p_bought.price,
+        (dexs.token_sold_amount_raw / power(10, erc20b.decimals)) * p_sold.price
+    )                                                     AS amount_usd,
+    dexs.token_bought_address,
+    dexs.token_sold_address,
+    COALESCE(dexs.taker, tx."from")                        AS taker,
+    dexs.maker,
+    dexs.project_contract_address,
+    dexs.tx_hash,
+    tx."from"                                              AS tx_from,
+    tx.to                                                  AS tx_to,
+    dexs.evt_index,
+    dexs.trace_address
+FROM dexs
+INNER JOIN {{ source('optimism', 'transactions') }} tx
+    ON dexs.tx_hash = tx.hash
+    {% if not is_incremental() %}
+    AND tx.block_time >= TIMESTAMP '{{project_start_date}}'
+    {% endif %}
+    {% if is_incremental() %}
+    AND tx.block_time >= date_trunc('day', now() - interval '7' day)
+    {% endif %}
+LEFT JOIN {{ ref('tokens_erc20') }} erc20a
+    ON erc20a.contract_address = dexs.token_bought_address
+    AND erc20a.blockchain = 'optimism'
+LEFT JOIN {{ ref('tokens_erc20') }} erc20b
+    ON erc20b.contract_address = dexs.token_sold_address
+    AND erc20b.blockchain = 'optimism'
+LEFT JOIN {{ source('prices', 'usd') }} p_bought
+    ON p_bought.minute = date_trunc('minute', dexs.block_time)
+    AND p_bought.contract_address = dexs.token_bought_address
+    AND p_bought.blockchain = 'optimism'
+    {% if not is_incremental() %}
+    AND p_bought.minute >= TIMESTAMP '{{project_start_date}}'
+    {% endif %}
+    {% if is_incremental() %}
+    AND p_bought.minute >= date_trunc('day', now() - interval '7' day)
+    {% endif %}
+LEFT JOIN {{ source('prices', 'usd') }} p_sold
+    ON p_sold.minute = date_trunc('minute', dexs.block_time)
+    AND p_sold.contract_address = dexs.token_sold_address
+    AND p_sold.blockchain = 'optimism'
+    {% if not is_incremental() %}
+    AND p_sold.minute >= TIMESTAMP '{{project_start_date}}'
+    {% endif %}
+    {% if is_incremental() %}
+    AND p_sold.minute >= date_trunc('day', now() - interval '7' day)
```

Comment:
Hi @ARDev097, you can use the `incremental_predicate` in incremental filters in future:
```suggestion
    AND {{ incremental_predicate('p_sold.minute') }}
```